### ISSUE TITLE
Occasional incorrect intial pair selection

### ISSUE THREAD

**Issue by donlk** [openMVG/openMVG#452]
Ok, i just bumped into this.

In sequential_SfM.cpp if automatic initial pair selection fails, the software calls for manual intervention. After a selection has been made, the reference parameter contains the desired initial pair, but it does not get passed for robust initial computation, as it happens for the automatically chosen pair. The local pair variable does not get overriden, resulting in occasional Segmentation Faults as tracks get selected between image indexes (0, 0), and the software assumes 2 unique index pairs in  MakeInitialPair3D(...).

Problematic section:
```
// Initial pair choice
  Pair initialPairIndex = _initialpair;
  if (_initialpair == Pair(0,0))
  {
    Pair putative_initial_pair;
    if (AutomaticInitialPairChoice(putative_initial_pair))
    {
      initialPairIndex = _initialpair = putative_initial_pair;
    }
    else // Cannot find a valid initial pair, try to set it by hand?
    {
      if (!ChooseInitialPair(_initialpair))
        return false;
    }
  }
  // Else a starting pair was already initialized before

  // Initial pair Essential Matrix and [R|t] estimation.
  if (!MakeInitialPair3D(initialPairIndex))
    return false;
```

Simple fix:
```
if (ChooseInitialPair(_initialpair)){
	initialPairIndex = _initialpair;
}else{
	return false;
}
```

**Comment by donlk** [comment]
Would you like a pull request?

**Comment by pmoulon** [comment]
Thanks for having found this bug.
You can perform a pull request on the develop branch.

The other way to see the problem can be to remove the initialPairIndex variable and only consider the _initialpair one.
So remove line https://github.com/openMVG/openMVG/blob/master/src/openMVG/sfm/pipelines/sequential/sequential_SfM.cpp#L96 and replace initialPairIndex by _initialpair on line https://github.com/openMVG/openMVG/blob/master/src/openMVG/sfm/pipelines/sequential/sequential_SfM.cpp#L113

**Comment by pmoulon** [comment]
Closed by merging #453
